### PR TITLE
(DOCSP-26078):  Flutter "User Management" landing page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Flutter "User Management" landing page

### Jira

- https://jira.mongodb.org/browse/DOCSP-26078

### Staged Changes (Requires MongoDB Corp SSO)

- [User Management (Flutter)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26078/sdk/flutter/users/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/users.txt [hunk 1]
```diff
@@ -11,3 +11,88 @@ Users - Flutter SDK
...
    Delete a User </sdk/flutter/users/delete-user>
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+To access Atlas App Services and use Atlas Device Sync with the Realm Flutter SDK,
+you must first authenticate a user with App Services Authentication.
```
Comment (dacharyc) [inline source/sdk/flutter/users.txt:22]: Should "Authentication" be capitalized in "App Services Authentication?" I don't think we tend to capitalize it in the App Services docs, or the other SDKs - I can't recall seeing it used like this as in "App Services Authentication." I can't find the naming doc at the moment to check if this is a proper feature name according to marketing. We might sidestep this by saying something like: "...you must first authenticate a user with an App Services authentication provider."

File: source/sdk/flutter/users.txt [hunk 2]
```diff
@@ -11,3 +11,88 @@ Users - Flutter SDK
...
+Atlas App Services documentation.
+
+.. _flutter-creating-and-deleting-users:
+
+Create and Delete Users
+-----------------------
+
+For all authentication providers other than email/password authentication,
+App Services automatically :ref:`registers a user <flutter-register>`
+the first time a user authenticates. If a user authenticates via more than
```
Comment (dacharyc) [inline source/sdk/flutter/users.txt:35]: I can see that you got this from existing User Mgt pages in other SDK docs, but seeing it here I am struck by the fact that we say "all auth providers other than email/password" but don't explain _why_ email/password is different. We might want to add a line here (and to any other SDK pages that have this text) that explicitly states something like:

> ...the first time a user authenticates. When using the email/password authentication provider, you manually `register` a user. 

(I imagine register might link to the SDK equivalent of the `registerUser` method, or maybe we just link `the email/password authentication provider` to the email/password auth page?)

File: source/sdk/flutter/users.txt [hunk 3]
```diff
@@ -11,3 +11,88 @@ Users - Flutter SDK
...
+- Refer to :ref:`User Objects <user-objects>` in the App Services documentation.
+- Refer to the Flutter SDK reference documentation about the
+  :flutter-sdk:`User <realm/User-class.html>` object.
+
+.. _flutter-app-work-with-custom-user-data:
+
+Custom User Data
+----------------
+
+You can associate custom data <custom-user-data with a user,
```
Comment (dacharyc) [inline source/sdk/flutter/users.txt:95]: Not sure if this was supposed to be a link or we removed the link, but `<customer-user-data` here needs a fixup.

File: source/sdk/flutter/users.txt [hunk 4]
```diff
@@ -11,3 +11,88 @@ Users - Flutter SDK
...
+users in and out <flutter-authenticate>` of your client app. You can:
+
+- Log users in with an existing social account, such as Apple, Facebook,
+  or Google.
+- Create new user accounts with App Services email/password management,
+  or your own custom function or custom JWT user management.
+- Enable anonymous users to let users access your App Services App without persisting
+  user data.
+
+When you have a logged-in user, you can:
```
Comment (dacharyc) [inline source/sdk/flutter/users.txt:60]: Should we add "Delete a user" to this list? I think with Swift anyway you need a logged-in `app.currentUser` to invoke the `delete` method.

Comment (mongodben) [reply source/sdk/flutter/users.txt:60]: absolutely!

File: source/sdk/flutter/users.txt (hunk 1 above)
Comment (mongodben) [reply source/sdk/flutter/users.txt:22]: i think technically it's supported to be capitalized but i like the sidestep suggestion

**Pull request merged**